src/raftstore/store/engine.rs

```diff
@@ -151,33 +151,43 @@ pub trait Peekable {
     }
 }
 
-#[derive(Clone)]
-pub enum SeekMode {
-    TotalOrderSeek,
-    PrefixSeek,
+#[derive(Clone, PartialEq)]
+enum SeekMode {
+    TotalOrder,
+    Prefix,
 }
 
-#[derive(Clone)]
 pub struct IterOption {
-    pub upper_bound: Option<Vec<u8>>,
-    pub fill_cache: bool,
-    pub seek_mode: SeekMode,
+    upper_bound: Option<Vec<u8>>,
+    fill_cache: bool,
+    seek_mode: SeekMode,
 }
 
 impl IterOption {
-    pub fn new(upper_bound: Option<Vec<u8>>, fill_cache: bool, seek_mode: SeekMode) -> IterOption {
+    pub fn new(upper_bound: Option<Vec<u8>>, fill_cache: bool) -> IterOption {
         IterOption {
             upper_bound: upper_bound,
             fill_cache: fill_cache,
-            seek_mode: seek_mode,
+            seek_mode: SeekMode::TotalOrder,
         }
     }
 
+    pub fn use_prefix_seek(mut self) -> IterOption {
```

Comment:
do we need another `use_total_order_seek` function if we want to change the seek mode dynamically?